File: ch2/py/e-rec-text-splitter-code.py
from langchain_text_splitters import (
    Language,
    RecursiveCharacterTextSplitter,
)

PYTHON_CODE = """ def hello_world(): print("Hello, World!") # Call the function hello_world() """

python_splitter = RecursiveCharacterTextSplitter.from_language(
    language=Language.PYTHON, chunk_size=50, chunk_overlap=0
)

python_docs = python_splitter.create_documents([PYTHON_CODE])

print(python_docs)

File: ch2/js/e-rec-text-splitter-code.js
import { RecursiveCharacterTextSplitter } from '@langchain/textsplitters';

const PYTHON_CODE = ` def hello_world(): print("Hello, World!") # Call the function hello_world() `;

const pythonSplitter = RecursiveCharacterTextSplitter.fromLanguage('python', {
  chunkSize: 50,
  chunkOverlap: 0,
});

const pythonDocs = await pythonSplitter.createDocuments([PYTHON_CODE]);

console.log(pythonDocs);
